espflash/src/cli/mod.rs

```diff
@@ -1187,6 +1168,20 @@ pub fn ensure_chip_compatibility(chip: Chip, elf: Option<&[u8]>) -> Result<()> {
     }
 }
 
+pub fn check_esp_idf_args(
+    format: ImageFormatKind,
+    erase_parts: &Option<Vec<String>>,
+    erase_data_parts: &Option<Vec<DataType>>,
+) -> Result<()> {
+    if format != ImageFormatKind::EspIdf && (erase_parts.is_some() || erase_data_parts.is_some()) {
+        return Err(miette::miette!(
+            "`erase-parts` and `erase-data` parts are only supported when using the `esp-idf` format."
+        ));
+    }
+
+    Ok(())
+}
+
```

Comment:
Creating a fake `Test` `ImageFormatKind` and using `espflash flash  espflash/tests/data/esp32c6 --monitor --format test --erase-parts nvs`  results in:
```
Error: 
  × `erase-parts` and `erase-data` parts are only supported when using the `esp-idf` format.
  
```